### PR TITLE
Test against the LTS as well as the weekly

### PR DESCRIPTION
**Pull request by jtnord**
As we need to run the ATH against LTS version when they are released we
should ensure that the ATH itself has no regressions when run against
it.

This allows us to actually identify issues in the LTS as opposed to the
ATH no longer being compatible with it.
This requires that the ATH be compatible with the weekly and the LTS
line, which should have been mostly done but was never tested in the PR
builder.

As we have recently removed the Java8 test this should not add any extra load onto ci.jenkins.io that was not present a few weeks ago but adding Mark for this as I know the ATH is relatively intensive for CI resources.

There is possible going to be some test failures here that need to be fixed too (so WIP until at least I have a report on failures)

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -27,37 +27,40 @@ if (needSplittingFromWorkspace) {
...
+                                  eval \$(vnc.sh)
+                                  java -version
 
-                                run.sh firefox latest -Dmaven.test.failure.ignore=true -DforkCount=1 -B
-                            """
-                        }
-                    }
-                }
-            }
+                                  run.sh firefox jenkinsUnderTest -Dmaven.test.failure.ignore=true -DforkCount=1 -B
```
Comment (MarkEWaite) [inline Jenkinsfile:57]: Does the `run.sh` script expand jenkinsUnderTest to be the correct value or is there something else happening here that somehow expands that String variable?
```suggestion
                                  run.sh firefox ${jenkinsUnderTest} -Dmaven.test.failure.ignore=true -DforkCount=1 -B
```

**Review by MarkEWaite — APPROVED** [review]
Approved but with a question that likely indicates I don't really understand the Pipeline DSL string expansion rules well enough.

File: Jenkinsfile [hunk 2]
```diff
@@ -27,37 +27,40 @@ if (needSplittingFromWorkspace) {
...
-                                // but not letting the build to fail will cause next build not to try those tests again.
-                                allowEmptyResults: true
-                        ) {
-                            sh """
-                                set-java.sh $javaVersion
-                                eval \$(vnc.sh)
-                                java -version
+      for (String v in ["lts", "latest"]) {
+          int javaVersion = j
+          String jenkinsUnderTest = v
```
Comment (timja) [inline Jenkinsfile:32]: what's the point to this? just name it jenkins under test in the loop same with java version?

File: Jenkinsfile (hunk 1 above)
Comment (timja) [reply Jenkinsfile:57]: 😆  nice catch

Comment (jtnord) [reply Jenkinsfile]: no I just messed up doing a quick patch in `notepad2`

File: Jenkinsfile (hunk 2 above)
Comment (jtnord) [reply Jenkinsfile:32]: https://www.jenkins.io/doc/pipeline/examples/#jobs-in-parallel

File: Jenkinsfile [hunk 3]
```diff
@@ -27,37 +27,40 @@ if (needSplittingFromWorkspace) {
...
+          def name = "java-${javaVersion}-jenkins-${jenkinsUnderTest}split${index}"
```
Comment (jtnord) [inline Jenkinsfile:33]:
```suggestion
          def name = "java-${javaVersion}-jenkins-${jenkinsUnderTest}-split${index}"
```

**Comment by jtnord** [comment]
well this sucks :-o
```
master51820|java.io.IOException: Failed to load: Form element path plugin (1.12)
master51820| - Jenkins (2.323) or higher required
master51820|	at hudson.PluginWrapper.resolvePluginDependencies(PluginWrapper.java:1016)
master51820|	at hudson.PluginManager$2$1$1.run(PluginManager.java:535)
master51820|	at org.jvnet.hudson.reactor.TaskGraphBuilder$TaskImpl.run(TaskGraphBuilder.java:169)
master51820|	at org.jvnet.hudson.reactor.Reactor.runTask(Reactor.java:296)
master51820|	at jenkins.model.Jenkins$5.runTask(Jenkins.java:1151)
master51820|	at org.jvnet.hudson.reactor.Reactor$2.run(Reactor.java:214)
master51820|	at org.jvnet.hudson.reactor.Reactor$Node.run(Reactor.java:117)
master51820|	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:68)
master51820|	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
master51820|	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
master51820|	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by jtnord** [comment]
a few failing tests against LTS :-(

ConfigFileProvider are known others look like regressions in the test framework against LTS.

<html>
<body>


Class | Duration | Fail  | (diff) | Skip | (diff) | Pass | (diff) | Total | (diff)
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
ConfigFileProviderTest | 23 min | 36 | +36 | 0 |   | 0 |   | 36 | +36
JobDslPluginTest | 7 hr 6 min | 9 | +9 | 0 |   | 603 | +603 | 612 | +612
StageViewTest | 2 hr 19 min | 8 | +8 | 0 |   | 136 | +136 | 144 | +144
ExternalWorkspaceManagerPluginTest | 1 hr 9 min | 3 | +3 | 0 |   | 51 | +51 | 54 | +54
WorkflowPluginTest | 3 hr 4 min | 3 | +3 | 0 |   | 123 | +123 | 126 | +126
DeclarativePipelineTest | 27 min | 1 | +1 | 0 |   | 17 | +17 | 18 | +18
GradlePluginTest | 1 hr 8 min | 1 | +1 | 0 |   | 71 | +71 | 72 | +72
XvncPluginTest | 1 hr 18 min | 1 | +1 | 0 |   | 53 | +53 | 54 | +54


</body>
</html>

**Comment by timja** [comment]
probably best to do this against the new LTS line instead as it shouldn't have any failures?

**Comment by jtnord** [comment]
> probably best to do this against the new LTS line instead as it shouldn't have any failures?

there should not be any failures against the existing LTS line, but it seems we have introduced regressions and I do not at the moment have time to look into fixing them , especially as the new LTS line will be out in a few weeks.

File: Jenkinsfile [hunk 4]
```diff
@@ -27,37 +27,41 @@ if (needSplittingFromWorkspace) {
...
+                                  // results to be empty to break the circle: build after removal executes one empty split
+                                  // but not letting the build to fail will cause next build not to try those tests again.
+                                  allowEmptyResults: true
+                          ) {
+                              sh """
+                                  set-java.sh $javaVersion
+                                  eval \$(vnc.sh)
+                                  java -version
+                              """ + (jenkinsUnderTest == "lts" ? " export FORM_ELEMENT_PATH_VERSION=1.11" : "") + 
+                              """
```
Comment (timja) [inline Jenkinsfile:57]:
```suggestion
```

**Pull request merged**